test/Microsoft.AspNet.Server.KestrelTests/SocketOutputTests.cs

```diff
@@ -199,6 +199,116 @@ public void WritesDontCompleteImmediatelyWhenTooManyBytesIncludingNonImmediateAr
         }
 
         [Fact]
+        public async Task OnlyWritesRequestingCancellationAreErroredOnCancellation()
```

Comment:
Note to self: We should add a test where a bunch of pending writes that get completed b/c of a failed write instead of a canceled token before merging.

I can do this myself since this will require some small modifications to MockLibuv (I know _uv_err_name and _uv_strerror will need to be implemented.)

There's also a pretty big chance that we might need to merge a more limited version of this PR that only ensures writes always get completed, and merge the CancellationToken related chances post-RC2.